apps/extension/src/ui/apps/dashboard/routes/Settings/Accounts/DeleteFolderModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { AccountsCatalogTree } from "@core/domains/accounts/types"
+import Dialog from "@talisman/components/Dialog"
+import { Modal } from "@talisman/components/Modal"
+import { ModalDialog } from "@talisman/components/ModalDialog"
+import { provideContext } from "@talisman/util/provideContext"
+import { api } from "@ui/api"
+import { useCallback, useEffect, useState } from "react"
+import { useTranslation } from "react-i18next"
+import styled from "styled-components"
+
+const useDeleteFolderModalProvider = () => {
+  const [name, setName] = useState<string | null>(null)
+  const [treeName, setTreeName] = useState<AccountsCatalogTree | null>(null)
+  const [isOpen, setIsOpen] = useState(false)
+
+  const open = useCallback((name: string, treeName: AccountsCatalogTree) => {
+    setName(name)
+    setTreeName(treeName)
+    setIsOpen(true)
+  }, [])
+  const close = useCallback(() => setIsOpen(false), [])
+
+  useEffect(() => {
+    close()
+  }, [close])
+
+  return {
+    name,
+    treeName,
+    isOpen,
+    open,
+    close,
+  }
+}
+
+export const [DeleteFolderModalProvider, useDeleteFolderModal] = provideContext(
+  useDeleteFolderModalProvider
+)
+
+export const DeleteFolderModal = () => {
+  const { t } = useTranslation("admin")
+  const { name, treeName, close, isOpen } = useDeleteFolderModal()
+
+  return (
+    <Modal open={isOpen}>
+      <ModalDialog title={t("Delete Folder")} onClose={close}>
+        {name !== null && treeName !== null && (
+          <DeleteFolder name={name} treeName={treeName} onConfirm={close} onCancel={close} />
+        )}
+      </ModalDialog>
+    </Modal>
+  )
+}
+
+const StyledDialog = styled(Dialog)`
```

Comment:
A styled component ! :eyes:

FYI I've deleted Dialog as part of Tailwind Rampage, I'll make necessary changes with merging.